src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -387,16 +389,35 @@ private void processMultipleInvalidLineFiles(ArmResponseBatchData armResponseBat
 
     }
 
-    private String getErrorCode(ArmResponseInvalidLineRecord invalidLineRecord1, ArmResponseInvalidLineRecord invalidLineRecord2) {
-        UploadNewFileRecord uploadNewFileRecord2 = readInputJson(invalidLineRecord2.getInput());
+    private void setInvalidLineErrorDescription(ArmResponseInvalidLineRecord record1, ArmResponseInvalidLineRecord record2,
+                                                ExternalObjectDirectoryEntity eod) {
+
+        String operation1 = getOperation(record1);
+        String operation2 = getOperation(record2);
 
-        String errorCode;
-        if (CREATE_RECORD.equalsIgnoreCase(uploadNewFileRecord2.getOperation())) {
-            errorCode = invalidLineRecord2.getExceptionDescription() + " " + invalidLineRecord1.getExceptionDescription();
+        StringBuilder errorDescription = new StringBuilder();
+        if (CREATE_RECORD.equalsIgnoreCase(operation2)) {
+            appendErrorDescription(errorDescription, operation2, record2);
+            appendErrorDescription(errorDescription, operation1, record1);
         } else {
-            errorCode = invalidLineRecord1.getExceptionDescription() + " " + invalidLineRecord2.getExceptionDescription();
+            appendErrorDescription(errorDescription, operation1, record1);
+            appendErrorDescription(errorDescription, operation2, record2);
         }
-        return errorCode;
+        eod.setErrorCode(errorDescription.toString());
+    }
+
+    private String getOperation(ArmResponseInvalidLineRecord record) {
+        UploadNewFileRecord uploadRecord = readInputJson(record.getInput());
+        if (uploadRecord != null) {
+            return uploadRecord.getOperation();
+        }
+        String operation = getOperationFromArmResponseFilesInputField(record.getInput());
+        return operation != null ? operation : "UNKNOWN";
+    }
+
+    private void appendErrorDescription(StringBuilder errorDescription, String operation, ArmResponseInvalidLineRecord record) {
```

Comment:
Entire class could benefit from unit testing

Reply:
Updated integration tests to account for new code

Reply:
added some unit tests